### PR TITLE
add OpenSSH serialization for ed25519 keys (#4808)

### LINKED ISSUE #4808: Exception when getting public_bytes of Ed25519PublicKey
I created an ed25519 key with: `ssh-keygen -t ed25519`.
This one I loaded successfully with cryptography 2.6.1. Dumping the loaded key however creates the following exception:

```
>>> print(key.__class__)
<class 'cryptography.hazmat.backends.openssl.ed25519._Ed25519PublicKey'>

>>> key.public_bytes(Encoding.OpenSSH, PublicFormat.OpenSSH)
Traceback (most recent call last):
  File "/usr/lib/python3.6/code.py", line 91, in runcode
    exec(code, self.locals)
  File "<console>", line 1, in <module>
  File "/home/richard/.local/lib/python3.6/site-packages/cryptography/hazmat/backends/openssl/ed25519.py", line 45, in public_bytes
    encoding, format, self, self._evp_pkey, None
  File "/home/richard/.local/lib/python3.6/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1838, in _public_key_bytes
    return self._openssh_public_key_bytes(key)
  File "/home/richard/.local/lib/python3.6/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1886, in _openssh_public_key_bytes
    assert isinstance(key, ec.EllipticCurvePublicKey)
AssertionError
```

### PR DESCRIPTION
**Pull request by bernhl**
Fixes #4808

### CODE REVIEW

**Comment by alex** [comment]
Thanks for sending the PR! Can you please take a look at the failing pep8 check.

**Comment by bernhl** [comment]
Done

File: src/cryptography/hazmat/backends/openssl/backend.py [hunk 1]
```diff
@@ -1885,6 +1885,13 @@ def _openssh_public_key_bytes(self, key):
                 ssh._ssh_write_mpint(parameter_numbers.g) +
                 ssh._ssh_write_mpint(public_numbers.y)
             )
+        elif isinstance(key, ed25519.Ed25519PublicKey):
+            _bytes = key.public_bytes(serialization.Encoding.Raw,
```
Comment (alex) [inline src/cryptography/hazmat/backends/openssl/backend.py]: Please give this a different name that doesn't start with an underscore

File: tests/hazmat/primitives/test_serialization.py [hunk 2]
```diff
@@ -1293,6 +1293,20 @@ def test_load_ssh_public_key(self, backend):
             b"N\x06G\xecV\xbc\x19\xaf\xc6<k\x07[\xc6"
         )
 
+    def test_public_bytes_openssh(self, backend):
+        ssh_key = (
+            b"ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIG2fgpmpYO61qeAxGd0wgRaN/E4"
+            b"GR+xWvBmvxjxrB1vG user@chiron.local"
```
Comment (alex) [inline tests/hazmat/primitives/test_serialization.py]: Drop the comment from this input and then is this variable in the assertion at the end

**Pull request merged**